### PR TITLE
Dh 4627/add support for all llm types

### PR DESCRIPTION
**Pull request by MohammadrezaPourreza**

### CODE REVIEW

**Comment by ishaan-jaff** [comment]
@MohammadrezaPourreza thanks for the PR

Noticed you have a hosted product too:
- How do you manage user based rate limits / costs ?
- Do you allow users to select their LLMs on the hosted product? (It sees like the option is not exposed to them in the demo here: https://www.dataherald.com/)

File: requirements.txt [hunk 1]
```diff
@@ -1,7 +1,7 @@
 dnspython==2.3.0
 fastapi==0.98.0
 httpx==0.24.1
-langchain==0.0.230
+langchain==0.0.286
```
Comment (jcjc712) [inline requirements.txt:4]: can you validate that this change doesn't affect db connections? I think we had a version issue with sqlalchemy and the db drivers

Comment (MohammadrezaPourreza) [reply requirements.txt:4]: @ishaan-jaff Hi again, 
We are currently experiencing a package dependency issue with the latest version of LangChain, which includes the LiteLLM module. When upgrading LangChain from version 0.0.230 to 0.0.286, we encountered problems with some of the SQLAlchemy database drivers failing to work. Currently, the error we encounter after updating LangChain and attempting to connect to Snowflake is as follows:
`You have an incompatible version of 'pyarrow' installed (13.0.0). Please install a version that adheres to the requirement: 'pyarrow<10.1.0,>=10.0.1; extra == "pandas"'`
Have you ever faced a similar issue?

**Comment by MohammadrezaPourreza** [comment]
@ishaan-jaff Hi Ishaan we merged the the LiteLLM integration with our engine in this PR: https://github.com/Dataherald/dataherald/pull/205 Thanks for your contribution to improve our tool

**Comment by ishaan-jaff** [comment]
@MohammadrezaPourreza this is awesome ! thanks for using LiteLLM. Any feedback on it so far ?

**Pull request closed without merging**